data-prepper-plugins/otel-proto-common/src/main/java/org/opensearch/dataprepper/plugins/otel/codec/OTelProtoStandardCodec.java

```diff
@@ -0,0 +1,1179 @@
+                    final String schemaUrl,
+                    AtomicInteger droppedCounter,
+                    final Integer exponentialHistogramMaxAllowedScale,
+                    final Instant timeReceived,
+                    final boolean calculateHistogramBuckets,
+                    final boolean calculateExponentialHistogramBuckets) {
+            List<Record<? extends Metric>> recordsOut = new ArrayList<>();
+            for (io.opentelemetry.proto.metrics.v1.Metric metric : metricsList) {
+                try {
+                    if (metric.hasGauge()) {
+                        recordsOut.addAll(mapGauge(metric, serviceName, ils, resourceAttributes, schemaUrl, timeReceived));
+                    } else if (metric.hasSum()) {
+                        recordsOut.addAll(mapSum(metric, serviceName, ils, resourceAttributes, schemaUrl, timeReceived));
+                    } else if (metric.hasSummary()) {
+                        recordsOut.addAll(mapSummary(metric, serviceName, ils, resourceAttributes, schemaUrl, timeReceived));
+                    } else if (metric.hasHistogram()) {
+                        recordsOut.addAll(mapHistogram(metric, serviceName, ils, resourceAttributes, schemaUrl, timeReceived, calculateHistogramBuckets));
+                    } else if (metric.hasExponentialHistogram()) {
+                        recordsOut.addAll(mapExponentialHistogram(metric, serviceName, ils, resourceAttributes, schemaUrl, exponentialHistogramMaxAllowedScale, timeReceived, calculateExponentialHistogramBuckets));
+                    }
+                } catch (Exception e) {
+                    LOG.warn("Error while processing metrics", e);
+                    droppedCounter.incrementAndGet();
+                }
+            }
+            return recordsOut;
+        }
+
+        private List<? extends Record<? extends Metric>> mapGauge(
+                                         io.opentelemetry.proto.metrics.v1.Metric metric,
+                                         String serviceName,
+                                         final Map<String, Object> ils,
+                                         final Map<String, Object> resourceAttributes,
+                                         final String schemaUrl,
+                                         final Instant timeReceived) {
+            return metric.getGauge().getDataPointsList().stream()
+                .map(dp -> JacksonStandardGauge.builder()
+                        .withUnit(metric.getUnit())
+                        .withName(metric.getName())
+                        .withDescription(metric.getDescription())
+                        .withStartTime(OTelProtoStandardCodec.getStartTimeISO8601(dp))
+                        .withTime(OTelProtoStandardCodec.getTimeISO8601(dp))
+                        .withServiceName(serviceName)
+                        .withValue(OTelProtoStandardCodec.getValueAsDouble(dp))
```

Comment:
I am not sure, that the standard gauge/sum/histogram data point should always use a double value. Maybe there should be a distinction between `value.int` and `value.double` as documented here: <https://github.com/opensearch-project/opensearch-catalog/blob/main/docs/schema/observability/metrics/metrics.md>

Reply:
Will investigate and address in next PR. If you know, definitive answer, please let me know.